src/utilities/strategy-helpers/smi-canary-deployment-helper.ts

```diff
@@ -186,33 +186,31 @@ function getTrafficSplitObject(kubectl: Kubectl, name: string, stableWeight: num
     if (!trafficSplitAPIVersion) {
         trafficSplitAPIVersion = kubectlUtils.getTrafficSplitAPIVersion(kubectl);
     }
-    const trafficSplitObjectJson = `{
+
+    return `{
         "apiVersion": "${trafficSplitAPIVersion}",
         "kind": "TrafficSplit",
         "metadata": {
-            "name": "%s"
+            "name": "${getTrafficSplitResourceName(name)}"
         },
         "spec": {
             "backends": [
                 {
-                    "service": "%s",
-                    "weight": "%sm"
+                    "service": "${canaryDeploymentHelper.getStableResourceName(name)}",
+                    "weight": "${stableWeight}"
                 },
                 {
-                    "service": "%s",
-                    "weight": "%sm"
+                    "service": "${canaryDeploymentHelper.getBaselineResourceName(name)}",
+                    "weight": "${baselineWeight}"
                 },
                 {
-                    "service": "%s",
-                    "weight": "%sm"
+                    "service": "${canaryDeploymentHelper.getCanaryResourceName(name)}",
+                    "weight": "${canaryWeight}"
                 }
             ],
             "service": "%s"

```

Comment:
Looks like this `%s` is still here but not getting formatted in anymore. I think it needs to be `${name}`.

Reply:
Good catch! Will fix